core/src/main/java/org/jruby/RubyClass.java

```diff
@@ -307,6 +309,49 @@ public int getVariableTableSizeWithExtras() {
         return new HashMap<String, VariableAccessor>(getVariableAccessorsForRead());
     }
 
+    /**
+     * Request that the listed ivars (no @ in name) have field storage when we are reified
+     * @param names
+     */
+    public synchronized void requestFieldStorage(String name, Class<?> fieldType,  Boolean unwrap, Class<?> toType) {
+        AdvancedFieldConfiguration config = new JavaClassConfiguration.AdvancedFieldConfiguration(name, fieldType, unwrap, toType);
+        if (reifiedClass != null)
+            requestFieldStorage(config);
+        else {
+            if (javaClassConfiguration.requestedStorageVariables == null)
+                javaClassConfiguration.requestedStorageVariables = new ArrayList<>();
+            javaClassConfiguration.requestedStorageVariables.add(config);
+        }
+    }
+
+    /**
+     * Actually requests field storage for the ivar once we are reified
+     */
+    private void requestFieldStorage(AdvancedFieldConfiguration config) {
+        try {
+            Class<?> fieldType = getReifiedClass().getField(config.name).getType();
+            if (fieldType != config.fieldType)
+                throw runtime.newTypeError("java_field " + config.name + " has incorrectly specified types for @ivar mapping");
+            
+            // by default, unwrap if it's not assignable to IRO
+            boolean unwrap = config.unwrap == null ? !fieldType.isAssignableFrom(IRubyObject.class) : config.unwrap.booleanValue();
+            Class<?> unwrapType = config.unwrapType == null ? fieldType : config.unwrapType;
+            if (unwrap && !fieldType.isAssignableFrom(unwrapType))
+                throw runtime.newTypeError("java_field " + config.name + " has incorrectly specified unwrap type for @ivar mapping");
+            
+            variableTableManager.getVariableAccessorForJavaMappedVar("@" + config.name,
+                    unwrap,
+                    fieldType,
+                    unwrapType,
+                    RubyObjectSpecializer.LOOKUP.findGetter(getReifiedClass(), config.name, fieldType),
+                    RubyObjectSpecializer.LOOKUP.findSetter(getReifiedClass(), config.name, fieldType));
```

Comment:
There are many places where getReifiedClass is called repeatedly... these should use a variable and call it once.